imports/api/partners/partners.js

```diff
@@ -160,10 +160,15 @@ if (Meteor.isServer) {
 
   Partners.after.update(function (userId, doc, fieldNames, modifier, options) {
     const Memberships = Mongo.Collection.get('memberships');
-    if (modifier.$set && modifier.$set.userId && modifier.$set.userId !== this.previous.userId) {
-      Memberships.update({ partnerId: doc._id }, { $set: { userId: modifier.$set.userId } }, { selector: { role: doc.role }, multi: true });
-    } else if (this.previous.userId && modifier.$unset && 'userId' in modifier.$unset) {
-      Memberships.update({ partnerId: doc._id }, { $unset: { userId: '' } }, { selector: { role: doc.role }, multi: true });
+    if (modifier.$set?.userId && modifier.$set.userId !== this.previous.userId) {
```

Comment:
`    const Memberships = Mongo.Collection.get('memberships');
    const diff = rusdiff.diff(_.pick(this.previous, 'userId'), _.pick(doc, 'userId'));
    Object.setByString(diff, '$set.accepted', false);
    Memberships.update({ partnerId: doc._id }, diff, { selector: { role: 'owner' }, multi: true });

`